### DISCUSSION TITLE
Power Control Functions

### DISCUSSION THREAD

**Discussion by WillCodeForCats** [WillCodeForCats/solaredge-modbus-multi#207] [General]
This discussion is for general stuff about power control related functions: storage (battery) control and site limits.

You may also want to read issues #136 and #137 which were part of the development process.

**Comment by paulorrockgit** [comment]
Hi,

I'm sorry if this is a newbie question. I can change the storage control mode and issue remote commands like charge or discharge though the overview page and thats great, thank you for doing this work. How do I find those entities (?) to be able to access them in a trigger ? I've looked through all the trigger options and I cant find the right 'action' for my 23:30 'trigger' e.g. change storage remote command mode to "charge from PV and AC"

Any advice or pointers please ?

Paul

**Reply by mpredfearn** [reply]
If you're trying to make an automation to automatically charge from AC at 11:30, then you would set your trigger to time at 11:30, and action to "call service", the service to  "select.select_option", the entity "select.solaredge_i1_storage_default_mode" or "select.solaredge_i1_storage_command_mode" and "Charge from Solar Power and Grid" as the value for "option".

**Reply by WillCodeForCats** [reply]
Device-based actions looks OK on my dev system:

<img width="445" alt="Screen Shot 2022-12-02 at 8 39 29 AM" src="https://user-images.githubusercontent.com/48533968/205341641-bbf4188b-8d79-4d26-844a-05fd065982c3.png">

Device based triggers look the same for me as well.

**Reply by paulorrockgit** [reply]
The device actions in the manual drop downs all work perfectly. However I've never done any automations before and this bit **' the service to "select.select_option" '** is the key thing I didn't know and couldn't see in the docs . 

Literally just installed the whole home automation setup just to be able to charge my battery overnight because Solaredge wont give us owners who have paid thousands for a battery the access to do it. Rant over ;-) . Thank you for your help and for being nice to a newbie .

**Reply by Peterjohnwood** [reply]
So we should use storage default mode rather than storage command mode?

**Comment by paulorrockgit** [comment]
Also does anyone know for certain if setting the storage control mode to "remote control" overrides any time of use policies that have been set up by solaredge or the installer ? They've set one up on my system to charge from midnight to 2am but I want to completely override that and control it all myself. I think it does but don't want to break stuff and create a clash.

**Reply by tonywheeler60** [reply]
I've certainly found no issues at all.  I have access to create and apply Time of Use profiles on the SE portal (in admin) and have found that setting my system to "Remote Control" over-rides any ToU.  The control offered by this integration work very well indeed - although I have found that the system will revert to the Storage Default Mode even with very high values set in "Solaredge I1 Storage Command Timeout".  It's as though the system ignores what you set (values in seconds) after a period.  I have got around this by adding additional automations to force it to reset the required settings every 2 hours - or as required.  Others seem to experience this too.

**Comment by Ashpork** [comment]
My installer lets me contact them to ask them to put a charge profile into the solaredge inverter. But that’s a pain in the rear by design, and I don’t want to. Id rather be in control, as I can see the next days forecast and decide if I want to charge or not.Might be worth seeing if your installer can help, but I enjoy the control. Seth, didn’t realise the device had a service, I had been controlling the entities directly.Sent from my iPhoneOn 2 Dec 2022, at 17:53, paulorrockgit ***@***.***> wrote:﻿
Also does anyone know if setting the storage control mode to "remote control" overrides any time of use policies that have been set up by solaredge or the installer ? They've set one up on my system to charge from midnight to 2am but I want to completely override that and control it all myself.

**Comment by anselmhuewe** [comment]
Hi @WillCodeForCats,
I would like to use the settings you present here, but I struggle finding them in HA. I have successfully set up the integration and enabled power control options and storage control, but I do not know how to set it to "Remote". Is it done via this integration (if yes, how?) or do I have to do it in the monitoring interface of solar edge (I can select options MSC / storage profiles etc. there, but 'remote' is not offered as an option)?
As a result (I guess!?), my HA interface looks like this, missing the actions and entities you describe:
![image](https://user-images.githubusercontent.com/21361834/211923050-56a40cff-5e2b-49dc-84ae-cff7898eba41.png)
Your help would be much appreciated!

**Reply by anselmhuewe** [reply]
update: Just noticed, that the sensor indicates Advanded Power Control is Off (and has always been), although the checkmarks are set in the configuration of the integration.
![image](https://user-images.githubusercontent.com/21361834/212182361-b603652b-61e5-4452-acce-cd64ca7fbc95.png)
![image](https://user-images.githubusercontent.com/21361834/212183043-201af321-7187-4a9e-934c-ba425b3838d5.png)
Don't get it. How can I enable it?

**Reply by WillCodeForCats** [reply]
Control functions are disabled by default, to use them you need to enable Power Control Options.

https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration#integration-configuration-options

**Reply by WillCodeForCats** [reply]
On the second config screen is storage control also enabled?

**Reply by WillCodeForCats** [reply]
It will be option "Speichersteuerung aktivieren" has to be enabled on the second screen.

**Reply by anselmhuewe** [reply]
Yes, it is enabled. That's why I don't understand it.

**Reply by dfein38347g** [reply]
I'm having the same issue. I did find the remote setting under the battery settings when connected to the inverter's AP. I set it to remote but I still do not see any power options in HA. I do have the problem of illegal address with the battery and so it does not show up in HA. Is that a requirement to see the power options?

**Reply by WillCodeForCats** [reply]
IllegalAddress means your inverter does not support batteries (or their control options) over modbus.

https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues#not-all-inverters-support-battery-data

**Reply by anselmhuewe** [reply]
My battery is showing up and all the b1 sensors are providing data.
I disabled and enabled the power control options. Enabling it shows up in the log (but still, according to the sensor state, the option remains disabled. Also note that the log reports some other (relevant???) errors.
![image](https://user-images.githubusercontent.com/21361834/212430501-9fd98594-8d58-4184-a418-ab1794a95395.png)

I also noticed in the logs that the sensor if often unavailable, and is disabled after each of these events.
![image](https://user-images.githubusercontent.com/21361834/212430411-725d0641-db9e-43e6-b039-ca04f6b51a36.png)

Do these infos help to debug it, @WillCodeForCats? Thanks a lot!

**Comment by ianfriend** [comment]
Hi all,

Apologies if i'm just being blind, but i'm looking to get some clarity on what the AC Charge Policy settings do in the power control options:

![image](https://user-images.githubusercontent.com/8939108/220071248-60353bf2-86ac-46f0-9ecf-d493612cdb76.png)

there doesn't seem to be any mention of these in the documentation.

TIA

**Reply by mpredfearn** [reply]
These just map directly to the Solaredge settings, so you'd be best off consulting the Solaredge documentation for the full description https://github.com/WillCodeForCats/solaredge-modbus-multi/blob/main/doc/Power-Control-Open-Protocol-for-SolarEdge-Inverters.pdf page 17.

**Reply by ianfriend** [reply]
ok thanks for the reply, that makes sense.
What i'm looking to do is to set a maximum charge % so my baterry wont charge past 90%.  I know i *could* do this with an automation, but was hoping the fixed energy limit policy with the AC charge limit would help achieve this, but turns out these are yearly limits, not dailiy limits.

**Comment by ianfriend** [comment]
Hi Guys,

So for ages now I keep getting continuous "unavailable" of sensors.  There doesnt seem to be a regular time when it does it, from this screenshot alone you can see its ranges from 4 mins to 49 mins.  What am i doing wrong?  My polling frequancy is 30 seconds and my interver delay 3 seconds.  i've played about a bit with these but doesnt seem to help.

![image](https://user-images.githubusercontent.com/8939108/234030391-b534b30c-5a62-463d-b011-a7ffcd122dfb.png)


TIA

**Reply by WillCodeForCats** [reply]
A 49 minute loss of communication sounds more like an inverter or network problem, not something that can be fixed by adjusting settings.

**Reply by ianfriend** [reply]
Thanks for the reply, but its not a loss of communication for that long, when it does "become unavailable" it comes back within 20 seconds or so , the 4-49 minutes i mentioned was how often it does it shown by the logs i screenshot. I pretty much always get this after i've changed a setting (changed storage control mode for example) but also does it at random other times.

**Reply by Ashpork** [reply]
How I’ve you got the inverter and HA connected to the router? Wired?

**Reply by ianfriend** [reply]
HA Wired, Inverter Wireless but with good signal

**Reply by Ashpork** [reply]
In the component logs do you see messages about 0 size packet received? I’ve seen this previously and eventually came down to network resilience despite Wi-Fi signal looking good. By wireless, you mean using the inverter wireless? Or a Wi-Fi extender back to eth?

**Reply by ianfriend** [reply]
Yes i do see 0 size packets in teh logs, as well as "can't open connection...".  My Inverter is in my detatched garage so i have a powerline from the main house network to a wifi hotspot in the garage. Its very possibly its to do with that netowrk setup, although i have other things attached to that hotspot too (AC / GYM equipment etc) and they all work fine, and speed tests show low latency / jitter and good speeds.
Just trying to work out a way of testing with ethernet as a test, but wanted to rule out a known issue from the wise heads here first :)

**Reply by Ashpork** [reply]
If you have an SSH terminal on your HA instance, what's it like when performing a ping to your inverter, if you want to try stressing it a little more, with the '-A' argument? Is there any loss? It might be the connection is resilient enough for a more modern protocol, making it seem like it's OK.

On paper, the advice is "go wired", which takes out most of the variance in the network. 

In reality, I am also using a WIFI repeater and dropping it back down to eth. and have great success with it, But I've had to set up the repeater to be located fairly close to my router (8m with one thick wall in the way), if i didn't do this, then I'd get lots of intermittent issues ranging from just one polling interval of being unavailable, through to 2 hours of the devices showing as unavailable. This for me is a stop gap, whilst I wait for a new broadband connection (FTTP) to be installed which would let me know which wall to put the eth through and ultimately remove the wifi repeater.

**Reply by WillCodeForCats** [reply]
The 2.3.0 pre-releases adds internal retry attempts to the HA update cycle that's worth a try.

**Reply by ianfriend** [reply]
> If you have an SSH terminal on your HA instance, what's it like when performing a ping to your inverter, if you want to try stressing it a little more, with the '-A' argument? Is there any loss? It might be the connection is resilient enough for a more modern protocol, making it seem like it's OK.
> 
> On paper, the advice is "go wired", which takes out most of the variance in the network.
> 
> In reality, I am also using a WIFI repeater and dropping it back down to eth. and have great success with it, But I've had to set up the repeater to be located fairly close to my router (8m with one thick wall in the way), if i didn't do this, then I'd get lots of intermittent issues ranging from just one polling interval of being unavailable, through to 2 hours of the devices showing as unavailable. This for me is a stop gap, whilst I wait for a new broadband connection (FTTP) to be installed which would let me know which wall to put the eth through and ultimately remove the wifi repeater.

Thanks for the detailed reply - useful info.  I've connected thew inverter directly to my house wifi, signal is weaker but cuts out the powerline adapter.  still the same though, normal usage with no long outages, jsut these 20 second "unavailable" every so often.

**Reply by ianfriend** [reply]
> The 2.3.0 pre-releases adds internal retry attempts to the HA update cycle that's worth a try.

Thanks, thats good to know, i'll give that a try

**Reply by ianfriend** [reply]
> If you have an SSH terminal on your HA instance, what's it like when performing a ping to your inverter, if you want to try stressing it a little more, with the '-A' argument? Is there any loss? It might be the connection is resilient enough for a more modern protocol, making it seem like it's OK.
> 
> On paper, the advice is "go wired", which takes out most of the variance in the network.
> 
> In reality, I am also using a WIFI repeater and dropping it back down to eth. and have great success with it, But I've had to set up the repeater to be located fairly close to my router (8m with one thick wall in the way), if i didn't do this, then I'd get lots of intermittent issues ranging from just one polling interval of being unavailable, through to 2 hours of the devices showing as unavailable. This for me is a stop gap, whilst I wait for a new broadband connection (FTTP) to be installed which would let me know which wall to put the eth through and ultimately remove the wifi repeater.

Just to add - i've run the ping from HA.  No fails but now and again it really slows down.  this could be the issue.  If its taking 14 seconds ro respond  then HA might time out (and therefore sensor become unavailable).  Its there a way of increasing the timeout?

![image](https://user-images.githubusercontent.com/8939108/234536609-eb2873ff-7534-4fb9-848b-8f8789b2dcbf.png)

**Reply by mpredfearn** [reply]
At the start of that high latency period, you lost ping sequence 1211, so your network is definitely lossy as well as suffering occasional high latency.

**Reply by ianfriend** [reply]
good spot, having now stopped the ping i can see that 16 of the 1482 pings were lost.  I'll see if i can figure out a way of getting ethernet to the inverter to run the same test.  Maybe the retry in the pre-release will serve as a workaround for now.

thanks all for your help here.

**Reply by ianfriend** [reply]
I've also jsut run this ping test with other devices in the garage, both connected to the powerline hotspot and the house wifi directly, and none of them suffer the same packet loss or latency slowdown.  I think this might be an issue with the inverter itself.

**Comment by davidecorradini** [comment]
Hello! Can you please tell me what is the difference between these two modes for Storage Control? Thanks

**Discharge to Minimize Import:**
If PV production < consumption, discharge battery only for self-consumption, not for grid export. Requires an external consumption/export meter.

**Maximize self consumption:**
Use PV production for self-consumption, then charge/discharge battery as needed to maximize self-consumption. Requires an external consumption/export meter.

**Reply by Ashpork** [reply]
Discharge to Minimize Import, will not charge your battery if you have
excess PV power not used by your load(home).

Maximize self consumption will prioritise meeting the needs of your
load(home) using PV, the excess PV will then go to the battery, and any excess after
that will go to the grid.

**Reply by davidecorradini** [reply]
Got it! Thank you for the prompt reply!

**Reply by mpredfearn** [reply]
The Solaredge documentation is quite good (https://github.com/WillCodeForCats/solaredge-modbus-multi/blob/main/doc/Power-Control-Open-Protocol-for-SolarEdge-Inverters.pdf page 17) but "Discharge to Minimize Import" will *only discharge* i.e. not charge the battery from excess solar, where "Maximize self consumption" will balance house load with solar generation, charging from excess and discharging to cover shortfall.

**Comment by petehhhhh** [comment]
Hi,

I am trying to use the power control to limit solar production during negative feed in tariffs. When I change the active power limit, it works for about 10 seconds then resets itself to 100%. Same thing happens when I change storage control modes. Suspect the Solaredge cloud is overriding my commands? I contacted my installer to ask if I could control locally. They contacted Solaredge support who said they do not support this function ? Any ideas as to how I can get this sorted ?

**Reply by deqw** [reply]
Did your installer add you already as a user with full access to your installation? Other option: send an e-mail to mysolaredgeapp@solaredge.com and request to join the beta program and ask for full access. I'm not sure if you need to have a home battery to be able to joint the beta. Be patient to watch your mailbox for what happens next (make take some time for them to react). When you suspect the SolarEdge cloud is overriding your commands maybe try what happens when you disable the internet to your inverter.

**Reply by petehhhhh** [reply]
Not sure. What will I see in portal if I do have it ? I did ask for admin access and they did something but didn’t seem to give me much more in terms of access. Its just the monitoring site and there is no ability to change time of use etc. The portal is https://monitoring.solaredge.com right - there is not some other admin site I am missing ?

To get full access what exactly should I ask them i? The support lady knows less than I do technically so unless I spell it out I tend to go around in circles…

yes tried blocking internet access but same thing happens?!

**Reply by deqw** [reply]
To join the beta just send the e-mail like I said. Provide them with your site ID and inverter serial number and they will pick it up someday. But for me the beta did not add useful features (screenshot Battery Mode). Better use this modbus integration and go the Remote Control way.
![Screenshot 2023-10-08 at 10 51 36](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/65849349/870e70ef-19f4-4daf-b41f-32f3f42da3ef)
![IMG_7429](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/65849349/b8a52707-ce41-4e2a-8a84-ed8eacb28069)

**Reply by seatrope** [reply]
@deqw When you say the "Remote route" I am one of the unlucky ones where the power control and battery control registers are not exposed.

I noticed that on my Setapp, there is a "Time of Use + Remote" option in the inverter, but it is GRAYED OUT.

![image](https://github.com/user-attachments/assets/3e7860c0-e2b1-4486-9ea3-93e5a3645530)

I opened a case with SE support and they told me this - not sure whether to believe it or not:
![image](https://github.com/user-attachments/assets/98e1a526-4a86-4303-95e8-3bf5245d0d14)

@deqw  or anyone - any ideas on this would be appreciated!

**Reply by deqw** [reply]
What I meant was: use this integration and set ‘Storage Control Mode’ to ‘Remote Control’ which then enables you to do your own automations.

**Reply by seatrope** [reply]
Got it. I was looking for some way to enable the "Storage Control mode" at all. Again, I'm one of those with inverters that do not have the Storage Control/power control or battery registers enabled. very frustrating :(

Thanks! Out of curiosity @deqw since you are able to access the proper registers on your inverter - if you look at the Energy Control section of your inverter in Setapp - are you able to select "Time of Use + Remote"?

**Reply by deqw** [reply]
I have to look that up in setapp. My screenshot somewhere above is from mySolarEdge app. When I check the Time of use option there the inverter is set to Remote Control by solarEdge logic, which makes sense. Actually since a week I’m running with the Time of use setting to see what happens. This is what SE is calling Solaredge One. I have to say it works supprisingly well so far. So at the moment all my power settings are controlled by SE.

**Reply by seatrope** [reply]
Hmm very interesting, the Solaredge One "AI-powered" stuff. That's a good use of AI i guess. 

Time of Use is grayed out for me on mySolaredge app "not available for your site configuration". Bastards!

**Reply by deqw** [reply]
I have checked setapp for you now. I have the same 'Time Of Use + Remote' option greyed out, but none of the other options are selected though. In mySolerEdge app 'Time of Use' is selected. When options are not available in your region it is probably by local regulations.

![IMG_8275](https://github.com/user-attachments/assets/ddf14467-3cda-49bb-b34a-324cdfdefea4)
![IMG_8276](https://github.com/user-attachments/assets/1f16db8c-bc6d-4f36-9afb-b4f5914b9610)

**Comment by souhl** [comment]
I am using power control settings to switch off battery usage when I charge my EV. Unfortunately from time to time I cannot switch back to Max Self Consumption. Combination is SE 7K RWS with BYD.

**Comment by dirixmjm** [comment]
My setup consists of only a single simple SE5K, no power meter nor battery.
I want to achieve simple on/off control depending on whether the energy price is positive or negative.
As I understand it, for this I don't need the "advanced power control", can someone please confirm?

I have been playing with the "active power limit" and "commit power settings". Most likely due to delays in activating the 0% and 100% state I don't really understand what is actually the correct steps to take. Should it be sufficient to change "active power limit" to 0% or 100% using an automation, or do I also need to activated "commit power settings" each time?

**Reply by dirixmjm** [reply]
> My setup consists of only a single simple SE5K, no power meter nor battery. I want to achieve simple on/off control depending on whether the energy price is positive or negative. As I understand it, for this I don't need the "advanced power control", can someone please confirm?
> 
> I have been playing with the "active power limit" and "commit power settings". Most likely due to delays in activating the 0% and 100% state I don't really understand what is actually the correct steps to take. Should it be sufficient to change "active power limit" to 0% or 100% using an automation, or do I also need to activated "commit power settings" each time?

Ok, setting "advanced power control", then "limit control mode" to "Production Control" and changing "Site Limit" works instant and consistent.

**Comment by joelegg** [comment]
I'm trying to work out what production control does and what systems it's for. Cantileman seems to not work on mine with site limit, export, import and zero site limit. But if I enable production control it seems to work

**Reply by Dillster70** [reply]
Did you sort it out?  

I have the same issue - 90% of the time I'll set Limit Control to Export Control - Import/Export, and put a number in for the Site Limit and it will reset back after a minute or so.  Sometime it actually works, but can't figure out why it does sometimes and not others.

I end up resorting to setting the Power Reduce to something like 4% instead.

**Reply by joelegg** [reply]
Are you sure you're not enrolled with Amber smart shift or anything or any other external control 

Unfortunately mine does not work still but it's not a big deal as it's only costing me a little bit because I can charge my car most of the time.

**Reply by Dillster70** [reply]
I am with Amber, but according to the app im still in the Device Link requested stage... have been for 2 months. Given up on that happening, happy to control it myself.

I don't have a car to charge unfortunately, so if I don't curtail it i'm often sending 6kW into the grid at -3c. It adds up quickly! I'm down 60c today - I should be 0 if not a little +ve as I exported in the morning.

**Reply by joelegg** [reply]
If something is changing it back, it's very likely to be Amber. You need to unenroll in order to control it. Even if it says in progress still.

**Reply by joelegg** [reply]
Yeah I get it. I wasn't able to curtail today at all in our EV was not at home so yeah minus $0.03 today yay.

**Reply by joelegg** [reply]
It took a long time for them to unenroll me. I had to end up getting solar edges to force unenrolled.

**Comment by nomisLH** [comment]
Hi, I am new to this so appoligies if i have posted to the wrong area or insufficient info.  
I have a solaredge 3phase (SE10k-RWB48) inverter and have installed SE Modbus multi in order to use Predbat with Home Assistant to manage energy. I have installed Modbus into HA and the only entity I get an 'unavailable' status is 'AC charge limit'. I have read through the documentation and cant see anything obvious i have missed in setup.  I spoken to Solaredge who cheched external control was setup correctly, that there were no network controls setup and they updated the firmware to the latest version. But AC charge limit is still 'unavailable.  Does anyone recognise this issue and any ideas of what to check or is it a known issue with this inverter.  Any help apreciated   Thank you

**Reply by WillCodeForCats** [reply]
Upload your diagnostic file so I can see what your settings are.

**Reply by nomisLH** [reply]
Thank you for taking a look at this, here is the file
[config_entry-solaredge_modbus_multi-01KKV0SYYP2FH57A1P7V2MQDWN(2).json](https://github.com/user-attachments/files/26938742/config_entry-solaredge_modbus_multi-01KKV0SYYP2FH57A1P7V2MQDWN.2.json)

**Reply by nomisLH** [reply]
Hi there, did you have a chance to look at the file, wodered if you had any ideas on what i might be doing wrong - thank for your help
